### DISCUSSION TITLE
Single training session that is split accross multiple process runs

### DISCUSSION THREAD

**Discussion by YoelShoshan** [optuna/optuna#4081] [Q&A] [answered]
Hi!
Thanks for the awesome library :)

I'm using a computation cluster that limits how long a single process can run, for example, only 24 hour.
A week of training of a single hyper params setup would take 7 such consecutive running sessions, with model progress stored in checkpoints between them.

Is there a way to do all of the following:

1. Do study.ask() on the first process that runs, and use `study.report(...)` to report intermediate values
2. On the 2nd process, somehow get access to the same trial and keep reporting intermediate values
3. do the same as in previous step up to the last (7th) process
4. use `trial.tell(...)` on the final process to report the trial final metric value

?

If not, any suggestion on how to handle such situation?
(please assume no access to any process which runs more than 24 hours)

**Comment by not522** [comment, marked as answer]
I think [this checkpoint example](https://github.com/optuna/optuna-examples/blob/46c822ce96e8ab53ba0b6113bd84eb4755032276/pytorch/pytorch_checkpoint.py) helps you.

**Reply by YoelShoshan** [reply]
@not522 Thank you!
